api/src/components/additionalInformation/controller.ts

```diff
@@ -0,0 +1,92 @@
+
+export const create = async (
+    event: I.AuthenticatedAPIRequest<
+        I.CreateAdditionalInformationBody,
+        undefined,
+        I.CreateAdditionalInformationPathParams
+    >
+): Promise<I.JSONResponse> => {
+    try {
+        const publicationVersion = await publicationVersionService.getById(event.pathParameters.id);
+
+        if (!publicationVersion) {
+            return response.json(404, {
+                message: 'This publication version does not exist'
+            });
+        }
+
+        // Ensure that the publication version is DRAFT
+        if (publicationVersion?.currentStatus !== 'DRAFT') {
+            return response.json(403, {
+                message: 'You can only add additional information to a draft publication'
+            });
+        }
+
+        // Ensure that current user owns this publication version
+        if (event.user.id !== publicationVersion.user.id) {
+            return response.json(403, {
+                message: 'You cannot add additional information to a publication version you do not own'
+            });
+        }
+
+        // Ensure URL is valid
+        if (!Helpers.validateURL(event.body.url)) {
+            return response.json(403, {
+                message: 'You must supply a valid URL'
+            });
+        }
+
+        // Create the additional information
+        const additionalInformation = await additionalInformationService.create(event.pathParameters.id, event.body);
+
+        return response.json(200, additionalInformation);
+    } catch (err) {
+        console.log(err);
+
+        return response.json(500, { message: 'Unknown server error' });
+    }
+};
+
+export const deleteAdditionalInformation = async (
+    event: I.AuthenticatedAPIRequest<undefined, undefined, I.DeleteAdditionalInformationPathParams>
+): Promise<I.JSONResponse> => {
+    try {
+        const publicationVersion = await publicationVersionService.getById(event.pathParameters.publicationVersionId);
+
+        // Check that the publication version exists
+        if (!publicationVersion) {
+            return response.json(404, {
+                message: 'This publication version does not exist.'
+            });
+        }
+
+        // Ensure that the publication version is a draft
+        if (publicationVersion.currentStatus !== 'DRAFT') {
+            return response.json(403, {
+                message: 'You cannot delete additional information from a publication version that is not a draft.'
+            });
+        }
+
+        if (event.user.id !== publicationVersion.user.id) {
+            return response.json(403, {
+                message: 'You do not have permission to delete additional information from this publication version.'
+            });
+        }
+
+        const additionalInformation = await additionalInformationService.deleteAdditionalInformation(
+            publicationVersion.id,
+            event.pathParameters.additionalInformationId
```

Comment:
Do we need a check to confirm additionalInformationId is valid (number, exists etc)? Got checks here for publicationVersionId, but nothing for additionalInfoId. 

Could also do with a test I reckon.

Reply:
Have added a check and test for this.